### PR TITLE
ref(access): Dataclass to represent (requires_sso, sso_is_valid) pairs

### PR DESCRIPTION
**Pull request by RyanSkonnord**
The `_sso_params` function will likely need to be extracted as a service interface method. As a preliminary step, represent its return value as a named type to improve eventual clarity.

### CODE REVIEW

File: src/sentry/auth/access.py [hunk 1]
```diff
@@ -95,8 +100,15 @@ class Access(abc.ABC):
     # would be based on the same scopes as API access so there's clarity in
     # what things mean
 
-    sso_is_valid: bool = False
```
Comment (corps) [inline src/sentry/auth/access.py:98]: I dont' see any changes to call sites or usages of these values; is this really not being used?

Comment (RyanSkonnord) [reply src/sentry/auth/access.py:98]: Replaced by a property.

Comment (corps) [reply src/sentry/auth/access.py:98]: I see, those two below.  Is `sso_state` intended as a public interface then?

Comment (corps) [reply src/sentry/auth/access.py:98]: I don't have a strong opinion but just curious, since it seems there will be two ways of accessing that data.

Comment (RyanSkonnord) [reply src/sentry/auth/access.py:98]: No strong opinion either. If I were doing it from scratch, I'd prefer to make `SsoState` the public interface. Maybe I'll make a case for replacing the boolean properties later. For now, I'll err on the side of making `_sso_state` private and we can change our minds if we want.

Comment (corps) [reply src/sentry/auth/access.py:98]: 👍 My only main concern is changing too many things at once.  Can focus one thing down at a time while making general interface improvements.

File: src/sentry/auth/access.py [hunk 2]
```diff
@@ -47,46 +47,51 @@ def get_permissions_for_user(user_id: int) -> FrozenSet[str]:
...
-        sso_is_valid = True
-        requires_sso = False
-    else:
-        if auth_provider.flags.allow_unlinked:
-            requires_sso = False
-            sso_is_valid = True
+        return _SSO_BYPASS
+
+    if auth_provider.flags.allow_unlinked:
+        return _SSO_BYPASS
```
Comment (markstory) [inline src/sentry/auth/access.py:71]: Nice use of early returns :clap:

**Review by markstory — APPROVED** [review]
Looks good to me.

**Pull request merged**